### ISSUE TITLE
Bug: Infrared Goggles in need of improvements?

### ISSUE THREAD

**Issue by Journeyman716** [jdmclark/jkgfxmod#26]
Hi,

I've noticed the Infrared Goggles don't exactly look great in-game with the gfx mod, for one everything looks really dark when they're on, compared when they're off. Another issue is that the goggles do not overlay any texture glows, eg. Nar Shaddaa's walls. 

I would put screenshots here, but currently it brings me to mention another bug: In-game screenshots taken via DF2's screenshot feature are broken with the gfx mod, only showing purely pink images in the exported BMP images.

**Comment by SirYodaJedi** [comment]
Try taking the screenshots either by using FRAPS or by printing the screen in windowed mode. 
The in-game screenshot function probably is confused by the injector. It would probably need to have the injector export the screenshots itself, rather than relying on the in-game functionality. This would be better as a separate issue thread, though, as it is probably less important than the IR Goggles working improperly.

**Comment by jdmclark** [comment]
Hi @swegmast 

Thanks for reporting this issue. I've investigated the IR goggles and discovered two root causes.

First, JkGfxMod is currently interpreting vertex colors with a gamma of 2.4. This is much too high. After some testing, I've found a fairly close match using gamma 1.8. My guess is the JK developers used a gamma of 1.8 while working on the game, which wasn't uncommon at the time.

![pre4](https://user-images.githubusercontent.com/1043153/58365179-a07bf200-7e7d-11e9-9aa0-1452aa6750ef.jpg)

In addition to improving the IR goggles, this change improves lighting in areas that were previously too dark, and also makes certain transparent surfaces more accurate to the original renderer.

I'll push these changes shortly.

Second, JK implements full screen color effects (like the IR goggles) by tinting all of the vertex colors in the entire scene. Unfortunately, this means JkGfxMod can't tint texture glows the way you'd expect: if IR goggles could tint glowing textures, they'd also be subject to level lighting. Then they wouldn't glow anymore.
